pool/connection_pool_test.go

```diff
@@ -130,6 +144,74 @@ func TestConnSuccessfully(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestConn_NoExecuteRole_Supported(t *testing.T) {
+	test_helpers.SkipIfWatchOnceUnsupported(t)
+
+	healthyServ := servers[0]
+
+	ctx, cancel := test_helpers.GetPoolConnectContext()
+	defer cancel()
+	connPool, err := pool.Connect(ctx,
+		[]pool.Instance{makeNoExecuteInstance(healthyServ, connOpts)})
+	require.Nilf(t, err, "failed to connect")
+	require.NotNilf(t, connPool, "conn is nil after Connect")
+
+	defer connPool.Close()
+
+	args := test_helpers.CheckStatusesArgs{
+		ConnPool:           connPool,
+		Mode:               pool.ANY,
+		Servers:            []string{healthyServ},
+		ExpectedPoolStatus: true,
+		ExpectedStatuses: map[string]bool{
+			healthyServ: true,
+		},
+	}
+
+	err = test_helpers.CheckPoolStatuses(args)
+	require.Nil(t, err)
+}
+
+func TestConn_NoExecuteRole_Unsupported(t *testing.T) {
+	watchOnceUnsupported, err := test_helpers.IsTarantoolVersionLess(3, 0, 0)
+	require.NoError(t, err)
+	if !watchOnceUnsupported {
+		t.Skip("Skipping test for Tarantool with watch once support")
+	}
```

Comment:
It's would be better to add a helper into `test_helpers` package: `test_helpers.SkipIfWatchOnceSupported(t)`.

Reply:
Updated.